hack/crd-gen/main.go

```diff
@@ -75,8 +73,12 @@ func crds() []*apiextv1beta1.CustomResourceDefinition {
 		if err != nil {
 			panic(err)
 		}
-		objs = append(objs, obj.(*apiextv1beta1.CustomResourceDefinition))
+
+		if version == obj.GetObjectKind().GroupVersionKind().GroupVersion().Version {
```

Comment:
This is really tough to get right. I _think_ `v1beta1` might actually be correct here, so long as we set the rest of the fields such that they're compatible with `v1`.

My reasoning is because if you run this patch without a Kubernetes cluster to just generate YAML (like for a GitOps workflow), you'll get an error.

```bash
velero install     --provider aws     --plugins velero/velero-plugin-for-aws:v1.0.0     --bucket $BUCKET     --backup-location-config region=$REGION     --snapshot-location-config region=$REGION     --secret-file $HOME/my-secret --dry-run -o yaml
An error occurred: error finding Kubernetes API server config in --kubeconfig, $KUBECONFIG, or in-cluster configuration: invalid configuration: no configuration has been provided
```

On `master`, the same command runs without error and outputs the YAML.

@skriss what are your thoughts currently?

Reply:
We definitely need to have a default version of the CRDs that we generate, if, as @nrb says, we're running in dry-run mode and don't necessarily have a cluster that we can connect to to get a server-preferred version.  I would say that default version, if discovery can't be reached/we're in dry-run mode, should be `v1beta1`, as it's most backwards-compatible.